src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -900,6 +942,25 @@ private IOperation CreateBoundConversionOperation(BoundConversion boundConversio
             BoundExpression boundOperand = boundConversion.Operand;
             if (boundConversion.ConversionKind == CSharp.ConversionKind.MethodGroup)
             {
+                if (boundConversion is { Type: FunctionPointerTypeSymbol { Signature: var signature }, Syntax: PrefixUnaryExpressionSyntax addressOfExpression })
+                {
+                    return new AddressOfOperation(
+                        new MethodReferenceOperation(
+                            signature.GetPublicSymbol(),
```

Comment:
>signature.GetPublicSymbol() [](start = 28, length = 27)

This doesn't look right. This should be the method the pointer will point to, I think.